Sources/Integration/VMTests.swift

```diff
@@ -305,6 +308,106 @@ extension IntegrationSuite {
+        let id = "test-fsnotify-events"
+
+        let bs = try await bootstrap()
+        let directory = try createFSNotifyTestDirectory()
+        let container = try LinuxContainer(id, rootfs: bs.rootfs, vmm: bs.vmm) { config in
+            config.process.arguments = ["/bin/sh", "-c", "sleep 30"]  // Keep container running
+            config.mounts.append(.share(source: directory.path, destination: "/mnt"))
+        }
+
+        try await container.create()
+        try await container.start()
+
+        // Get the vminitd agent to send notifications
+        let connection = try await container.dialVsock(port: 1024)  // Default vminitd port
+        let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        let agent = Vminitd(connection: connection, group: group)
+
+        // Calculate the hashed tag name for the mount source and mount in VM
+        let mountTag = try hashMountSource(source: directory.path)
+        let vmMountPath = "/tmp/fsnotify-test"
+
+        try await agent.mount(.init(type: "virtiofs", source: mountTag, destination: vmMountPath))
+
+        // Test 1: CREATE event on existing file
+        let createResponse = try await agent.notifyFileSystemEvent(path: "\(vmMountPath)/existing.txt", eventType: .create)
+
+        guard createResponse.success else {
+            throw IntegrationError.assert(msg: "CREATE event failed: \(createResponse.error)")
+        }
+
+        // Test 2: MODIFY event on existing file
+        let modifyResponse = try await agent.notifyFileSystemEvent(path: "\(vmMountPath)/existing.txt", eventType: .modify)
+        guard modifyResponse.success else {
+            throw IntegrationError.assert(msg: "MODIFY event failed: \(modifyResponse.error)")
+        }
+
+        // Test 3: Verify inotify events are actually generated using inotifywait
+        let inotifyBuffer = BufferWriter()
+        let inotifyProcess = try await container.exec("test-inotify") { config in
+            // Install inotify-tools and monitor the mount point for events
+            config.arguments = [
+                "/bin/sh", "-c",
+                """
+                apk add --no-cache inotify-tools > /dev/null 2>&1 && \
+                timeout 2 inotifywait -m /mnt -e modify,create,delete --format '%e %f' 2>/dev/null &
+                INOTIFY_PID=$!
+                sleep 0.1
+                # Trigger a modify event that should be detected
+                touch /mnt/test-inotify.txt
+                echo "modify test-inotify.txt"
+                wait $INOTIFY_PID 2>/dev/null || true
+                """,
+            ]
+            config.stdout = inotifyBuffer
+        }
+
+        try await inotifyProcess.start()
+
+        // While inotify is running, send FSNotify events that should trigger inotify
+        try await Task.sleep(for: .milliseconds(200))
+        let _ = try await agent.notifyFileSystemEvent(path: "\(vmMountPath)/test-inotify.txt", eventType: .modify)
```

Comment:
I would expect that we'd start the inotify process, then run tests 1 and 2, and expect to see two modify events on `existing.txt`.